packages/base/src/3dview/mainview.tsx

```diff
@@ -1173,12 +1224,12 @@ export class MainView extends React.Component<IProps, IStates> {
   private _updateTransformControls(selection: string[]) {
     if (selection.length === 1) {
       const selectedMeshName = selection[0];
-      const matchingChild = this._meshGroup?.children.find(child =>
+      this._matchingChild = this._meshGroup?.children.find(child =>
         child.name.startsWith(selectedMeshName)
       );
 
-      if (matchingChild) {
-        this._transformControls.attach(matchingChild as BasicMesh);
+      if (this._matchingChild) {
+        this._transformControls.attach(this._matchingChild as BasicMesh);
```

Comment:
I'm surprised you're still attaching the matchingchild to the transform control. If you can can make it work with attaching the matchingchild, isn't the pivot object useless?

Reply:
for rotation the controls have been attaching to pivot only [see here](https://github.com/jupytercad/JupyterCAD/pull/574/files/e78d8c8954513076a5c529ee0ed9e7c6f2807dbb#diff-37a7f3f4d02e2bc4c4f40a612b75d24569d7c0c46420298e3281f4af6e29b7e6R1252)

For translation i've kept the logic straight and same as before